decompressor/lz4/lz4.go

```diff
@@ -0,0 +1,95 @@
+package lz4
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pierrec/lz4"
+	"io"
+)
+
+type Reader struct {
+	src  io.Reader
+	data []byte
+}
+
+func NewReader(src io.Reader) *Reader {
+	return &Reader{src: src}
+}
+
+var lz4Type = []byte{0x82}
+
+func (r *Reader) Read(buf []byte) (n int, err error) {
+	// exhaust remaining data from previous Read()
+	if len(r.data) > 0 {
+		n = copy(buf, r.data)
+		r.data = r.data[n:]
+		if len(r.data) == 0 {
+			r.data = nil
+		}
+		return
+	}
+
+	// checksum
+	b := make([]byte, 16)
+	n, err = r.src.Read(b)
```

Comment:
Use [io.ReadFull](https://golang.org/pkg/io/#ReadFull) instead of `Read` here, since `Read` may read less than `len(b)` bytes.